### DISCUSSION TITLE
(Wrong) language of place names?

### DISCUSSION THREAD

**Discussion by dl9rdz** [photoprism/photoprism#3698] [Q&A]
I noted that Photoprism tags fotos with place names in apparently Arabic (the location is in Iceland, not an Arabic speaking country)

https://places.photoprism.app/v1/location/48D60B316530
return name "تمثال حورية البحر الصغيرة"

I presume that OSM is the source for the database used by the places lookup AP?

In above example, most likely the source is this node https://www.openstreetmap.org/node/3808515513/history
which contains an name_ar entry with exactly above text, but it also contains an "int_name" entry "The Little Mermaid" and a "name" entry "Hafmeyjan".  Why is Photoprism using the Arabic name in this case, not the normal local name or international name?

**Comment by lastzero** [comment]
I can't tell you without doing further research. However, this could take a while, as we are busy with other tasks at the moment...

Our reverse geocoding backend queries and merges information from two different OSM indexes, which should generally return the local (native) name for locations - unless maybe it's not available (in the version of the index we have installed) or their locale isn't properly set.

**Comment by lloyd-dada** [comment]
I would love to help improve this API lookup so that the user can set the preferred language to get from OSM.

**Comment by qroac** [comment]
> ... which should generally return the local (native) name for locations - unless maybe it's not available (in the version of the index we have installed) or their locale isn't properly set.

I' love to have an option to set a preferred language for location names, maybe even a kind of priority list like:
Use german, if not available use english, else use native or whatever else

Usecase: 
I just came across this discussion because I started importing my photo collection into photoprism.
For our diving trips in Thailand, the reverse geocoding api set these location:
`บ้านหาดทรายรี, จังหวัดสุราษฎร์ธานี, Thailand` 
My Thai is not that good, so the location names in Thai are quite useless for me ;)
I would really prefer to have the location names in german or at least in english
